libs/deposits/src/lib/deposit-form.tsx

```diff
@@ -151,7 +151,11 @@ export const DepositForm = ({
         <Controller
           control={control}
           name="asset"
-          rules={{ validate: { required } }}
+          rules={{
+            validate: {
+              required: (value) => !!selectedAsset || required(value),
```

Comment:
@mattrussell36  I do need this check here. the asset can be selected and the value would still be undefined. defaultValue set up is not sufficent.